portfolio/src/main/webapp/index.html

```diff
@@ -6,7 +6,7 @@
     <link rel="stylesheet" href="style.css">
     <script src="script.js"></script>
   </head>
-  <body>
+  <body onload="fetchBlobstoreUrl()">
```

Comment:
Optional:
Does the blobstore URL have a TTL (Time to live)? Meaning: does it expire at some time and you can't use it anymore? In this case, you would need to refresh the URL when the user clicks on upload.